packages/block-editor/src/store/reducer.js

```diff
@@ -810,40 +817,45 @@ export const blocks = flow(
 					},
 				};
 
-			case 'UPDATE_BLOCK_ATTRIBUTES':
-				// Ignore updates if block isn't known
-				if ( ! state[ action.clientId ] ) {
+			case 'UPDATE_BLOCK_ATTRIBUTES': {
+				// Avoid a state change if none of the block IDs are known.
+				if ( action.clientIds.every( ( id ) => ! state[ id ] ) ) {
 					return state;
 				}
 
-				// Consider as updates only changed values
-				const nextAttributes = reduce(
-					action.attributes,
-					( result, value, key ) => {
-						if ( value !== result[ key ] ) {
-							result = getMutateSafeObject(
-								state[ action.clientId ],
-								result
-							);
-							result[ key ] = value;
-						}
-
-						return result;
-					},
-					state[ action.clientId ]
+				const next = action.clientIds.reduce(
+					( accumulator, id ) => ( {
+						...accumulator,
+						[ id ]: reduce(
```

Comment:
We could use the vanilla `[].reduce` here; we're already using it a few lines above.

Reply:
But it's not an array that we reduce.